src/main/java/school/hei/haapi/authentication/AwsCognitoJwtIdToken.java

```diff
@@ -0,0 +1,65 @@
+package school.hei.haapi.authentication;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.proc.ConfigurableJWTProcessor;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Collection;
+import java.util.List;
+@Component
+public class AwsCognitoJwtIdToken {
+  @Autowired
+  private JwtConfiguration jwtConfiguration;
+  @Autowired
+  private ConfigurableJWTProcessor configurableJWTProcessor;
+  private static final Log logger = LogFactory.getLog(AwsCognitoJwtIdToken.class);
+  private Collection<GrantedAuthority> getAuthorities (){
+    return List.of(new SimpleGrantedAuthority("ROLE_USER"), new SimpleGrantedAuthority("ROLE_ADMIN"));
+  }
+
+  public Authentication authenticate(HttpServletRequest request) throws Exception {
+    String idToken = request.getHeader(this.jwtConfiguration.getHttpHeader());
+    if (idToken != null) {
+      JWTClaimsSet claims =
+              this.configurableJWTProcessor
+                      .process(this.getBearerToken(idToken),null);
+      validateIssuer(claims);
+      verifyIfIdToken(claims);
+      String username = getUserNameFrom(claims);
+      if (username != null) {
+        User user = new User(username, "", getAuthorities());
+        return new JwtAuthentication(user, claims, getAuthorities());
+      }
+    }
+    return null;
+  }
+  private String getUserNameFrom(JWTClaimsSet claims) {
+    return claims.getClaims()
+            .get(this.jwtConfiguration.getUserNameField())
+            .toString();
+  }
+
+  private void verifyIfIdToken(JWTClaimsSet claims) throws Exception {
+    if (!claims.getIssuer().equals(this.jwtConfiguration.getCognitoIdentityPoolUrl())) {
+      throw new Exception("Not an ID Token");
```

Comment:
1. Never use bare Exception or bare RuntimeException. Always extends them and use more specific exceptions.
2. Add an interceptor for handling in a uniform way the mapping from interal exception to external Error Model. See ExceptionMapper (return ApiException in the different mapping methods)